prime-dotnet-webapi/Services/DocumentService.cs

```diff
@@ -76,5 +76,13 @@ public async Task<string> FinalizeDocumentUpload(Guid documentGuid, string fileP
         {
             return await _documentManagerClient.FinalizeUploadAsync(documentGuid, filePath);
         }
+
+        public async Task<string> GetDownloadTokenForHealthAuthorityOrgAgreementDocument(int healthAuthorityId)
+        {
+            var document = await _context.HealthAuthorities
+                .Include(ha => ha.HealthAuthorityOrganizationAgreementDocument)
+                .SingleAsync(ha => ha.Id == healthAuthorityId);
```

Comment:
using `SingleAsync` means an item is expected however it looks like `GetDownloadTokenForHealthAuthorityOrgAgreementDocument` does not necessarily always need to return a document token since organization agreement is optional. Maybe `SingleOrDefaultAsync` is more appropriate here.